packages/@aws-cdk/aws-servicecatalog/lib/portfolio.ts

```diff
@@ -66,6 +70,17 @@ export interface IPortfolio extends cdk.IResource {
    * @param options Options for the initiate share
    */
   shareWithAccount(accountId: string, options?: PortfolioShareOptions): void;
+
+  /**
+   * Associate portfolio with the given product.
+   * @param product A service catalog produt.
+   */
+  addProduct(product: IProduct): void;
+
+  /**
+   * Allow or disallow tag updates, which adds a Resource Update Constraint.
+   */
+  allowTagUpdates(product: IProduct, options?: TagUpdateConstraintOptions): void;
```

Comment:
How about an alternative design for this?

`allowTagUpdates` takes only a `CommonConstraintOptions`, and always set the flag to `true`, and there's a separate `disallow/disableTagUpdates()` (exact name up to you of course) method that set the flag to `false`.

Basically, I think having a method called `allowTagUpdates()` which can actually do the exact opposite of what it's called, depending on the arguments passed to it, is weird.

Reply:
I think having them in the same method actually makes things clearer and explicit.  Because the underlying generated template still only creates one L1 resource for this.  It would seem weird to have different methods that update and affect the same underlying resource.  Especially since the default behavior for this is false (when it's not added) I think it would raise confusion since one would assume they wouldn't need a disallow at all.

Reply:
I don't think having a method called `allowTagUpdates()` that can be used to disallow tag updated is very clear and explicit...

Reply:
I am fine with calling this `constrainTagUpdates` and have renamed it as such

Reply:
Leave it as `allowTagUpdates()` if that's what you like more, just change the name of the property inside `TagUpdateConstraintOptions` to `allow`, so that it's 100% clear it's the same "allow" as in the name of the method: `portfolio.allowTagUpdates(product, { allow: false })`.